### PR TITLE
 #408 Removed use of deprecated ClientConfiguration

### PR DESCRIPTION
**Pull request by mikewalch**
Still might do a little more work on this. I am putting this up for review and to have Jenkins run ITs

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java [hunk 1]
```diff
@@ -35,7 +36,7 @@
   /**
    * @return Zookeeper connection information for Accumulo instance
    */
-  String getZookeepers();
+  String getZooKeepers();
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java]: Capital K?

File: core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      } else if (key.equals(ClientProperty.SSL_USE_JSSE.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_USE_JSSE, val);
+      } else if (key.equals(ClientProperty.SASL_ENABLED.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_RPC_SASL_ENABLED, val);
+      } else if (key.equals(ClientProperty.SASL_QOP.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SASL_QOP, val);
+      } else if (key.equals(ClientProperty.SASL_KERBEROS_SERVER_PRIMARY.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.KERBEROS_SERVER_PRIMARY, val);
+      } else {
+        config.setProperty(key, val);
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java]: Switch case might be cleaner

File: core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java (hunk 1 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java]: That's how we have referenced them in the past and how Apache ZooKeeper spells it. I can switch back but @ctubbsii might disagree :-)

File: core/src/main/java/org/apache/accumulo/core/client/impl/ConnectionInfoImpl.java [hunk 3]
```diff
@@ -38,7 +39,7 @@ public String getInstanceName() {
   }
 
   @Override
-  public String getZookeepers() {
+  public String getZooKeepers() {
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/impl/ConnectionInfoImpl.java]: Do we have a consistent way to refer to "zookeepers"?  Camel case or one word?

File: core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java (hunk 1 above)
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java]: If it's public API, we should follow upstream naming conventions. Internally, it doesn't really matter, other than as best practice for developing the habit of using the trademark properly.

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java]: Probably best for another PR

**Comment by mikewalch** [comment]
I pushed another commit ef0c46bca5b231d4 to fix issues with the shell. Can you take a look at it?

File: core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java (hunk 2 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java]: I think they both are kind of ugly.  I would prefer to keep it as is.

File: core/src/main/java/org/apache/accumulo/core/client/impl/ConnectionInfoImpl.java (hunk 3 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/impl/ConnectionInfoImpl.java]: While I would prefer `Zookeeper`, `ZooKeeper` is how the project refers to itself.

**Comment by mikewalch** [comment]
I refactored `ClientConfConverter` in 04576f18e6c68.  I am ready to merge if everyone approves.

**Comment by milleruntime** [comment]
:+1:

**Pull request merged**